### PR TITLE
Followup edits to 4847

### PR DESCRIPTION
**Pull request by bfallonf**
For #4847 

cc @pecameron let me know if you have any thoughts.

@openshift/team-documentation PTAL

### CODE REVIEW

File: architecture/core_concepts/routes.adoc [hunk 1]
```diff
@@ -917,19 +917,18 @@ endpoint, the service's weight is distributed among the endpoints with each endp
...
 `weight` 0, requests are returned with a 503 error. 
 
 When using `alternateBackends` also use the `roundrobin` xref:load-balancing[load balancing strategy]
 to ensure requests are distributed as expected to the services based on `weight`.
 `roundrobin` can be set for a route using a xref:route-specific-annotations[route annotation],
 or for the router in general using an xref:env-variables[environment variable].
 
-Here is an example of using alternate backends for
+The following is an example route configuration using alternate backends for
 xref:../../dev_guide/deployments/advanced_deployment_strategies.adoc#advanced-deployment-a-b-deployment[a/b deployments].
```
Comment (ahardin-rh) [inline architecture/core_concepts/routes.adoc]: A/B

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 2]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
 
-Use a xref:dev_guide/deployments/deployment_strategies.adoc#[blue-green deployment]
-when you want to switch users to a new, blue, version of your application in
-a production, green, environment.
+[NOTE]
+====
+Since many applications depend on persistent data, you will need to have an
+application that supports xref:n1-compatibility[N-1 compatibility] if you share
+a database, or implement a live data migration between your database, store, or
+disk if you choose to create two copies of your data layer.
```
Comment (mburke5678) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: This paragraph is not clear to me. "you need to either:
* have an application that supports n-1
*implement live migration
??

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 3]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
 
-Blue-green deployments make switching between two different versions of your
-application easy. However, since many applications depend on persistent data,
-you will need to have an application that supports xref:n1-compatibility[N-1
-compatibility] if you share a database, or implement a live data migration
-between your database, store, or disk if you choose to create two copies of your
-data layer.
+Consider the data used in testing the new version. If it is the production data
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: data, setting

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 4]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
-a bug in the new version can break the production version.
 
 [[advanced-deployment-strategies-blue-green-deployments-example]]
-=== Blue-Green Deployment Example
+=== Using a Blue-Green Deployment
 
-This strategy uses two deployment configurations. One, green, is for the current,
-production, version and the other is for the new, blue, version. Each deployment
-configuration is exposed to a different service.
+Blue-Green deployment use two deployment configurations. Both are running, and
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: deployments

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 5]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
+the one that is used depends on the service the route specifies, with each
+deployment configuration exposed to a different service. You can create a new
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: Both are running and exposed to a different service.

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 6]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
+version and the other for the newer, blue, version. 
 
-This example sets up two deployment configurations, one is for the stable, green,
-version and the other for the new, blue, version.  Since a route points to a
-service, and can be changed to point to a different service at any time. As a
-developer you can test the new version of your code by connecting to
-the new service before your production traffic is routed to it. Routes are
-intended for web (HTTP and HTTPS) traffic, so this technique is best suited
-for web applications.
+Since a route points to a service, and can be changed to point to a different
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: This is an incomplete sentence

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 7]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
+service at any time. As a developer you can test the new version of your code by
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: developer, you

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 8]
```diff
@@ -115,36 +115,34 @@ endif::[]
...
+environment. You can specify that the production version gets most of the user
+requests while a limited fraction of requests go to the new version. Since you
+control the portion of requests to each version, as testing progresses you can
+increase the fraction of requests to the new version and ultimately stop using
+the previous version. As you adjust the request load on each version, the number
+of pods in each service may need to be scaled as well to provide the expected
+performance.
+
+In addition to upgrading software, this feature can be used to experiment with
+versions of the user interface. Since some users get the old version and some
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: you can use this feature to experiment

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 9]
```diff
@@ -115,36 +115,34 @@ endif::[]
...
+the new, you can evaluate the user's reaction to the different versions to
+inform design decisions.
+
+For this to be effective both the old and new versions need to be similar enough
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: effective, both

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 10]
```diff
@@ -115,36 +115,34 @@ endif::[]
...
+that both can run at the same time. This is common with bug fix releases and
+when new features don't interfere with the old. The versions need
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: do not

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 11]
```diff
@@ -153,92 +151,72 @@ The user sets up a xref:../../architecture/core_concepts/routes.adoc#alternateBa
...
-however existing connections remain active.  When the service `*weight*` is not 0,
-each endpoint has a minimum `*weight*` of 1. Because of this a service with a lot
-of endpoints can end up with higher `*weight*` than desired. In this case reduce
-the number of pods to get the desired load balance `*weight*`.
+service is the `*service_weight*` divided by the `*sum_of_weights*`. The
+`*weight*` for each service is distributed to the service's endpoints so that
+the sum of the endpoint `*weights*` is the service `*weight*`.
+
+The route can have up to four services. The `*weight*` for the service can be
+between 0 and 256. When the `*weight*` is 0 no new requests go to the service,
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: values in back-ticks?

, no new requests

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 12]
```diff
@@ -153,92 +151,72 @@ The user sets up a xref:../../architecture/core_concepts/routes.adoc#alternateBa
...
+however existing connections remain active. When the service `*weight*` is not
+0, each endpoint has a minimum `*weight*` of 1. Because of this a service with a
```
Comment (ahardin-rh) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: this, a service

In this case,

File: dev_guide/deployments/deployment_strategies.adoc [hunk 13]
```diff
@@ -52,28 +51,28 @@ downtime may be needed to complete the transition.
 - If you have a non-isolated test environment, you can break both new and old versions.
 
 
-Since the end user usually accesses the application through a route handled by a 
-router. The deployment strategy can focus on deployment configuration features or 
-routing features.
+Since the end user usually accesses the application through a route handled by a
```
Comment (ahardin-rh) [inline dev_guide/deployments/deployment_strategies.adoc]: This is an incomplete sentence. Combine the two?

router, the deployment strategy can focus

File: dev_guide/deployments/deployment_strategies.adoc [hunk 14]
```diff
@@ -52,28 +51,28 @@ downtime may be needed to complete the transition.
...
-application. Strategies that use router features target individual routes.
+Strategies that focus on the deployment configuration impact all routes that use
+the application. Strategies that use router features target individual routes.
 
-Many deployment strategies are supported through the deployment configuration and
-some additional strategies are supported through router features. The deployment
-configuration based strategies are discussed in this section.
+Many deployment strategies are supported through the deployment configuration
+and some additional strategies are supported through router features. The
+deployment configuration based strategies are discussed in this section.
```
Comment (ahardin-rh) [inline dev_guide/deployments/deployment_strategies.adoc]: configuration-based

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 15]
```diff
@@ -28,51 +30,49 @@ endif::[]
...
+If necessary, you can roll back to the older, green, version by switching
+service back to the previous version.
 
+[discrete]
 [[advanced-deployment-strategies-using-a-route-and-two-services]]
-=== Using a Route and Two Services
+==== Using a Route and Two Services
+
+This example sets up two deployment configurations; one for the stable, green,
+version and the other for the newer, blue, version.
```
Comment (mburke5678) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: Maybe "the stable version (green) and the other for the newer version (blue). A bit awkward as is, to me anyway.

File: dev_guide/deployments/advanced_deployment_strategies.adoc [hunk 16]
```diff
@@ -115,36 +115,34 @@ endif::[]
...
+versions of the user interface. Since some users get the old version and some
+the new, you can evaluate the user's reaction to the different versions to
+inform design decisions.
+
+For this to be effective both the old and new versions need to be similar enough
+that both can run at the same time. This is common with bug fix releases and
+when new features don't interfere with the old. The versions need
+xref:n1-compatibility[N-1 compatibility] to work together properly.
+
+{product-title} supports this through the web console as well as the command
```
Comment (mburke5678) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: What is "this"?

File: dev_guide/deployments/deployment_strategies.adoc [hunk 17]
```diff
@@ -12,22 +12,21 @@ toc::[]
...
+is to make the change without downtime in a way that the user barely notices the
 improvements.
 
-The most common strategy is to use a blue-green deployment where the software is 
-just switched from the current, green, version to the new, blue, version for all
-users. The new,blue, version is brought up for testing and evaluation while the
-users still use the previousi, green, version.  When ready, the users are
-switched to the new, blue, version. If a problem arises just switch back to
+The most common strategy is to use a xref:../../dev_guide/deployments/advanced_deployment_strategies.adoc#advanced-deployment-strategies-blue-green-deployments[blue-green deployment]. The new, blue, version is brought up for testing and evaluation while the
+users still use the stable, green, version. When ready, the users are
```
Comment (mburke5678) [inline dev_guide/deployments/deployment_strategies.adoc]: should we use _green_ and _blue_ as these are a first reference?

File: dev_guide/deployments/deployment_strategies.adoc (hunk 14 above)
Comment (mburke5678) [inline dev_guide/deployments/deployment_strategies.adoc]: How about "Strategies based on deployment configurations are discussed in this section."

**Comment by mburke5678** [comment]
@bfallonf I added a few comments.

File: dev_guide/deployments/advanced_deployment_strategies.adoc (hunk 2 above)
Comment (bfallonf) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: This was in there already, but reworded.

File: dev_guide/deployments/advanced_deployment_strategies.adoc (hunk 6 above)
Comment (bfallonf) [inline dev_guide/deployments/advanced_deployment_strategies.adoc]: Oops bad edit.

**Comment by bfallonf** [comment]
Thanks @ahardin-rh @mburke5678 I made the suggestions. Will merge.

**Comment by bfallonf** [comment]
[rev_history]
|xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Core Concepts -> Routes]
|Added the xref:../architecture/core_concepts/routes.adoc#alternateBackends[Alternate Backends] section, and various information about multiple deployments.
%
|xref:../dev_guide/deployments/advanced_deployment_strategies.adoc#dev-guide-advanced-deployment-strategies[Deployments -> Advanced Deployment Strategies]
|Added information about running multiple deployments.
%
|xref:../dev_guide/deployments/deployment_strategies.adoc#dev-guide-deployment-strategies[Deployments -> Deployment Strategies]
|Added information about running multiple deployments.
%

**Comment by gaurav-nelson** [comment]
@ahardin-rh I have updated the labels and milestone for weekly release.

**Pull request merged**